lib/utils/getAdditionalEntries.js

```diff
@@ -0,0 +1,61 @@
+const querystring = require('querystring');
+
+/** @typedef {{ prependEntries: import('./injectRefreshEntry').WebpackEntry, overlayEntries: import('./injectRefreshEntry').WebpackEntry }} AdditionalEntries */
+
+/**
+ * Creates an object that contains two entry arrays: the prependEntries and overlayEntries
+ * @param {Object} optionsContainer This is the container for the options to this function
+ * @param {import('../types').NormalizedPluginOptions} optionsContainer.options Configuration options for this plugin.
+ * @param {import('webpack').Compiler.options.devServer} [optionsContainer.devServer] The webpack devServer config
+ * @returns {AdditionalEntries} An object that contains the Webpack entries for prepending and the overlay feature
+ */
+function getAdditionalEntries({ options, devServer }) {
+  /** @type {Record<string, *>} */
+  let resourceQuery = {};
+
+  if (devServer) {
+    const { sockHost, sockPath, sockPort, port, host } = devServer;
+
+    (sockHost || host) && (resourceQuery.sockHost = sockHost ? sockHost : host);
+    sockPath && (resourceQuery.sockPath = sockPath);
+    (sockPort || port) && (resourceQuery.sockPort = sockPort ? sockPort : port);
+  }
+
+  if (options.overlay) {
+    options.overlay.sockHost && (resourceQuery.sockHost = options.overlay.sockHost);
+    options.overlay.sockPath && (resourceQuery.sockPath = options.overlay.sockPath);
+    options.overlay.sockPort && (resourceQuery.sockPort = options.overlay.sockPort);
+    options.overlay.sockProtocol && (resourceQuery.sockProtocol = options.overlay.sockProtocol);
```

Comment:
We can actually also derive the value of this from the `https` option in `devServer`

Reply:
so up above in the `if(devServer)` block, I should add a line like:

```js
https && (resourceQuery.sockProtocol = 'https');
```

that look right?

Reply:
```js
resourceQuery.sockProtocol = https ? 'https' : 'http';
```

I think dev server defaults to http?

Reply:
oh hmm it says that it accepts an object for https as well, but doesn't give an example of what that object would look like

any ideas?

https://webpack.js.org/configuration/dev-server/#devserverhttps

Reply:
This should be it - while looking at the docs I also realised that the user could set `http2` which would default to `https` too.

[Ref](https://github.com/webpack/webpack-dev-server/blob/master/lib/Server.js)

```js
this.options.https = {
  ca: this.options.ca,
  pfx: this.options.pfx,
  key: this.options.key,
  cert: this.options.cert,
  passphrase: this.options.pfxPassphrase,
  requestCert: this.options.requestCert || false,
  /** @deprecated */
  spdy: {
    protocols: ['h2', 'http/1.1'],
  },
};
```

Reply:
ok so if it's an object we can just assume we need https then it seems.

so it should be?

```js
resourceQuery.sockProtocol = https || http2 ? 'https' : 'http';
```

Reply:
Yup exactly!